mantidproject/mantid
# Allow PaalmanPingsMonteCarloAbsorption to use shape defined in workspace

**Description of work.**

Previously PPMCA would require the user to define a new sample (and possibly container) shape before running, however there is now the option to use the shape preset in the workspace data.
If the preset shape is being used, there is still the option to override the sample material without affecting the sample and can geometries. 
As part of this, the algorithm now only takes one input workspace for both sample and container rather than one for each.
The Indirect Corrections Calculate Monte Carlo tab that uses this algorithm is updated to support these changes.

**To test:**
This test script runs the algorithm a few times using the preset shape, and once using a preset shape but overridden material.
[testscript.zip](https://github.com/mantidproject/mantid/files/5703743/testscript.zip)
1. Try this script out, test using some different shapes/parameters/materials etc
2. Try the Corrections interface (Interfaces -> Indirect -> Corrections) to ensure this reflects the updated functionality

Fixes #29899 .

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
